src/main/java/net/dries007/tfc/objects/items/ceramics/ItemSmallVessel.java

```diff
@@ -282,16 +288,41 @@ public void addHeatInfo(@Nonnull ItemStack stack, @Nonnull List<String> text)
             else
             {
                 boolean hasContent = false;
+                Map<Metal, Integer> materials = new HashMap<>();
+                boolean onlySmeltables = true;
                 for (ItemStack slot : super.stacks)
                 {
                     if (!slot.isEmpty())
                     {
+                        IMetalItem itemMetal = CapabilityMetalItem.getMetalItem(slot);
+                        if (itemMetal != null)
+                        {
+                            materials.merge(itemMetal.getMetal(slot), itemMetal.getSmeltAmount(slot) * slot.getCount(), Integer::sum);
+                        }
+                        else
+                        {
+                            onlySmeltables = false;
+                        }
                         text.add(1, I18n.format(TerraFirmaCraft.MOD_ID + ".tooltip.small_vessel_item", slot.getCount(), slot.getItem().getItemStackDisplayName(slot)));
                         hasContent = true;
                     }
                 }
 
-                if (!hasContent)
+                if (hasContent)
+                {
+                    if (onlySmeltables)
+                    {
+                        int textPosition = (int) super.stacks.stream().filter(itemstack -> !ItemStack.EMPTY.equals(itemstack)).count() + 1;
```

Comment:
What is `super.stacks` doing here this looks very confusing to me? (This is not a criticism I actually don't understand what's going on here)

Reply:
That gets the count of non-empty item stacks so we can place the tooltip right after the contents text, since that can be between 0-4 and super.stacks.size() always returns 4.

Now that I think about it, it probably should be `super.stacks.stream().filter(slot->!slot.isEmpty()).count()` instead.